portal/src/graphql/adminapi/AddGroupScreen.tsx

```diff
@@ -1,7 +1,206 @@
+          />
+          <WidgetDescription className="mt-2">
+            <FormattedMessage id="AddGroupScreen.groupName.description" />
+          </WidgetDescription>
+        </div>
+        <div>
+          <FormTextField
+            required={true}
+            fieldName="key"
+            parentJSONPointer=""
+            type="text"
+            label={renderToString("AddGroupScreen.groupKey.title")}
+            value={formState.groupKey}
+            onChange={onFormStateChangeCallbacks.groupKey}
+          />
+          <WidgetDescription className="mt-2">
+            <FormattedMessage id="AddGroupScreen.groupKey.description" />
+          </WidgetDescription>
+        </div>
+        <FormTextField
+          multiline={true}
+          resizable={false}
+          autoAdjustHeight={true}
+          rows={3}
+          fieldName="description"
+          parentJSONPointer=""
+          type="text"
+          label={renderToString("AddGroupScreen.groupDescription.title")}
+          value={formState.groupDescription}
+          onChange={onFormStateChangeCallbacks.groupDescription}
+        />
+      </RoleAndGroupsVeriticalFormLayout>
+      <RoleAndGroupsFormFooter className="mt-12">
+        <PrimaryButton
+          disabled={!canSave || isUpdating}
+          type="submit"
+          text={<FormattedMessage id="create" />}
+        />
+        <DefaultButton
+          disabled={isUpdating}
+          type="button"
+          onClick={cancel}
+          text={<FormattedMessage id="cancel" />}
+        />
+      </RoleAndGroupsFormFooter>
+    </div>
+  );
+}
 
 const AddGroupScreen: React.VFC = function AddGroupScreen() {
-  return <></>;
+  const breadcrumbs = useMemo<BreadcrumbItem[]>(() => {
+    return [
+      {
+        to: "~/user-management/groups",
+        label: <FormattedMessage id="GroupsScreen.title" />,
+      },
+      { to: ".", label: <FormattedMessage id="AddGroupScreen.title" /> },
+    ];
+  }, []);
+
+  const navigate = useNavigate();
+
+  const { createGroup } = useCreateGroupMutation();
+
+  const validate = useCallback((rawState: FormState): APIError | null => {
+    const [_, errors] = validateGroup({
+      key: rawState.groupKey,
+      name: rawState.groupName,
+      description: rawState.groupDescription,
+    });
+    if (errors.length > 0) {
+      return makeLocalValidationError(errors);
+    }
+    return null;
+  }, []);
+
+  const submit = useCallback(
+    async (rawState: FormState) => {
+      const [sanitizedRole, errors] = validateGroup({
+        key: rawState.groupKey,
+        name: rawState.groupName,
+        description: rawState.groupDescription,
+      });
+      if (errors.length > 0) {
+        throw new Error("unexpected validation errors");
+      }
+      return createGroup({
+        key: sanitizedRole.key,
+        name: sanitizedRole.name,
+        description: sanitizedRole.description,
+      });
+    },
+    [createGroup]
+  );
```

Comment:
We have the requirement of auto fill the key field as name if key is not provided

https://linear.app/authgear/issue/DEV-404/roles-and-groups-configuration-in-portal#comment-e5e223c7

Reply:
I should have done it in [8f4e839](https://github.com/authgear/authgear-server/pull/3966/commits/8f4e83908a1ba3c9629a9d937276f46ea5ce894b)

Reply:
But I found that the variable name here is wrong due to copy and paste error. Let me fix it.